state/remoteapplication.go

```diff
@@ -50,6 +53,29 @@ type remoteEndpointDoc struct {
 	Scope     charm.RelationScope `bson:"scope"`
 }
 
+type attributeMap map[string]interface{}
+
+// RemoteSpace represents a space in another model that endpoints are
+// bound to.
+type RemoteSpace struct {
```

Comment:
Is there a reason why this isn't remoteSpaceDoc?
And remoteSubnetDoc below?
To match what's done for remoteEndpointDoc.
And there'd be a separate struct used for public use outside of state package. We want to layer mongo/state data structs separately from structs used outside of state. That's either done via embedding the fooDoc in a Foo struct, or using a help func to map between the unexported state struct doc and the exported public struct.